src/app/(auth)/subscriber/edit/page.tsx

```diff
@@ -0,0 +1,67 @@
+'use client';
+
+import Subscriber from '@/components/Subscriber/Subscriber';
+import { UserGroupIcon } from '@heroicons/react/24/outline';
+import { updateSubscriber, getSubscriber } from '@/apis/subscriber';
+import { useEffect, useState } from 'react';
+import { toast } from 'react-toastify';
+import { useRouter, useSearchParams } from 'next/navigation';
+
+const Page = () => {
+  const searchParams = useSearchParams();
+  const id = searchParams.get('id');
+  const router = useRouter();
+  const [selectedOptions, setSelectedOptions] = useState([]);
+  const [subscriber, setSubscriber] = useState({});
+
+  useEffect(() => {
+    const fetchSubscriber = async () => {
+      try {
+        const { data } = await getSubscriber(id);
+        setSelectedOptions(
```

Comment:
how the options need to render on the form should be a responsibility of the form and not the edit page

Reply:
Here we are fetching the corresponding tags of each subscriber and then set them in a state. And only showing those selected tags in multi-select at the time of edit. That's why I add this to the edit page.